dbms/src/Storages/StorageDistributed.cpp

```diff
@@ -410,6 +410,9 @@ Pipes StorageDistributed::read(
                     throw Exception(exception_message.str(), ErrorCodes::UNABLE_TO_SKIP_UNUSED_SHARDS);
             }
         }
+
+        if (settings.force_optimize_skip_unused_shards_no_nested)
+            settings.force_optimize_skip_unused_shards = 0;
```

Comment:
This is slightly incorrect, because it will override the default value on remote node (because the setting will be `changed`).
Look at `dbms/src/Interpreters/ClusterProxy/executeQuery.cpp` for the details.
Maybe it's a better place to apply this setting.

Reply:
Updated, thanks!